### PR TITLE
[#9][Documentation] Document on Github Wiki the infrastructure

### LINKED ISSUE #9: [Documentation] Document on Github Wiki the infrastructure provisioning details, the system requirements, and setup instructions.
## Why

Keeping a record of how the infrastructure is set up is important for a bunch of good reasons. Firstly, it's handy for team members who need to set up or take care of the infrastructure. It also comes in handy when we're bringing new team members on board because it gives them a clear and complete guide to getting everything up and running.

On top of that, documenting the infrastructure setup is a big help in making sure the system stays reliable. Having detailed documentation makes it more likely that we'll get the infrastructure set up right each time, which means fewer mistakes or problems down the line.

## Acceptance Criteria
- The documentation should include details about the infrastructure provisioning process, including any tools or scripts used to set up the infrastructure.
- The documentation should specify the system requirements for the infrastructure, including any required software or hardware.
- The documentation should provide clear and detailed instructions for setting up the infrastructure, including any necessary configuration steps.
- The documentation should be organized in a clear and logical manner, with sections and headings as appropriate.
- The documentation should be suitable for sharing with other stakeholders, such as clients or upper management.

### PR DESCRIPTION
**Pull request by sanG-github**
- Close #9

## What happened 👀

Add the documentation that helps to know comprehensively how to set up the infrastructure for our application. That covered the need to define the `local variables` as well as `input variables`. 

Apart from that, also mention the `instance type` and all the default values that will be set for the resources.

## Insight 📝

`n/a`

## Proof Of Work 📹

| Documentation that will be pushed                                                                                             |
|-------------------------------------------------------------------------------------------------------------------------------|
| image |

### CODE REVIEW

File: .github/wiki/Architecture-Documentation.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+This document outlines the software and hardware requirements for running the infrastructure of the [example app](https://github.com/SanG-github/nimble-devops-ic-web). You can refer to the [[Architecture Diagram]] for a visual overview of the infrastructure components.
+
+## Required Hardware
+
+The infrastructure is hosted on AWS and relies on various AWS services, including ECS with Fargate launch type, RDS, and ElastiCache. Migrating to a different cloud provider is possible, thanks to Terraform's flexibility.
+
+### RDS Instance Type
+
+- Instance Type: `db.t3.micro`
```
Comment (longnd) [inline .github/wiki/Architecture-Documentation.md]: isn't it `db.t3.medium`? https://github.com/sanG-github/nimble-devops-ic-infrastructure/pull/33/files#diff-e350683dbd845c963c09473e02536035959202976a5bfe47a152c7f88fa7365aR42

File: .github/wiki/Architecture-Documentation.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+This document outlines the software and hardware requirements for running the infrastructure of the [example app](https://github.com/SanG-github/nimble-devops-ic-web). You can refer to the [[Architecture Diagram]] for a visual overview of the infrastructure components.
+
+## Required Hardware
+
+The infrastructure is hosted on AWS and relies on various AWS services, including ECS with Fargate launch type, RDS, and ElastiCache. Migrating to a different cloud provider is possible, thanks to Terraform's flexibility.
```
Comment (longnd) [inline .github/wiki/Architecture-Documentation.md]: this infrastructure project target AWS, so I believe if the following statement is correct
> Migrating to a different cloud provider is possible, thanks to Terraform's flexibility.

**Review by longnd — APPROVED** [review]

File: .github/wiki/Architecture-Documentation.md (hunk 1 above)
Comment (sanG-github) [reply .github/wiki/Architecture-Documentation.md]: I forgot to update the document after changing it to the new instance type 😪
It was updated in d5040a1 ✅ Thanks for your reminder 🙏🏻

File: .github/wiki/Architecture-Documentation.md (hunk 2 above)
Comment (sanG-github) [reply .github/wiki/Architecture-Documentation.md]: I haven't tried it yet, but those are my thoughts about Terraform 😗 Thus let me remove them to reduce confusion.


79bdf25 ✅

File: .github/wiki/Architecture-Documentation.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+### Required Variables
+
+All variables should be set as Terraform variables, and the 'sensitive' flag should be applied where necessary. The following variables are shared between all workspaces:
+
+| Variable                     | Value       |
+|:-----------------------------|-------------|
+| AWS_ACCESS_KEY_ID            | SENSITIVE   |
+| AWS_SECRET_ACCESS_KEY        | SENSITIVE   |
+| bastion_allowed_ip_connection| SENSITIVE   |
```
Comment (Nihisil) [inline .github/wiki/Architecture-Documentation.md]: `bastion_allowed_ip_connections` is set in `locals.tf` file, not in tf variables

Comment (sanG-github) [reply .github/wiki/Architecture-Documentation.md]: It was removed in 09f3d67 ✅

**Review by Nihisil — APPROVED** [review]

**Pull request merged**